tests/integration/tabular/test_base.py

```diff
@@ -150,4 +152,66 @@ def test_conditional_sampling_properly_handles_constraints(gm_mock):
     assert len(sample_calls) == 2
     model._model.sample.assert_any_call(5, conditions=expected_transformed_conditions)
     model._model.sample.assert_any_call(1, conditions=expected_transformed_conditions)
-    assert sampled_data.equals(expected_data)
+    pd.testing.assert_frame_equal(sampled_data, expected_data)
+
+
+@patch('sdv.tabular.copulas.copulas.multivariate.GaussianMultivariate',
+       spec_set=GaussianMultivariate)
+def test_conditional_sampling_constraint_uses_columns_model(gm_mock):
+    """Test that the ``sample`` method handles constraints with conditions.
+
+    The ``sample`` method is expected to properly apply constraint
+    transformations by sampling the missing columns for the constraint
+    if ``fit_columns_model`` is True.
+
+    Setup:
+    - The model is being passed a ``UniqueCombination`` constraint and then
+    asked to sample with two conditions, one of which the constraint depends on.
+    The constraint will sample the columns it needs that are not present in
+    the conditions and will then use constraint transformations to meet the
+    requirements.
+
+    Input:
+    - Conditions
+    Side Effects:
+    - Correct columns to condition on are passed to underlying sample method
+    """
+    # Setup
+    constraint = UniqueCombinations(
+        columns=['city', 'state'],
+        handling_strategy='transform'
+    )
+    data = pd.DataFrame({
+        'city': ['LA', 'SF', 'CHI', 'LA', 'LA'],
+        'state': ['CA', 'CA', 'IL', 'CA', 'CA'],
+        'age': [27, 28, 26, 21, 30]
+    })
+    model = GaussianCopula(constraints=[constraint], categorical_transformer='label_encoding')
+    sampled_numeric_data = [pd.DataFrame({
+        'city#state': [0, 1, 2, 0, 0],
+        'age': [30, 30, 30, 30, 30]
+    }), pd.DataFrame({
+        'city#state': [1],
+        'age': [30]
+    })]
+    gm_mock.return_value.sample.side_effect = sampled_numeric_data
+    model.fit(data)
+
+    # Run
+    conditions = {'age': 30, 'state': 'CA'}
+    sampled_data = model.sample(5, conditions=conditions)
+
+    # Assert
+    expected_states = pd.Series(['CA', 'CA', 'CA', 'CA', 'CA'], name='state')
+    expected_ages = pd.Series([30, 30, 30, 30, 30], name='age')
+    sample_calls = model._model.sample.mock_calls
+    assert len(sample_calls) == 2
+    assert sample_calls[0][1][0] == 5
+    assert sample_calls[1][1][0] == 1
+    assert sample_calls[0][2]['conditions']['age'] == 30
+    assert sample_calls[1][2]['conditions']['age'] == 30
+    assert 'city#state' in sample_calls[0][2]['conditions']
```

Comment:
I do not understand this assert. Would `city#state` be expected to NOT be inside the `conditions` being passed to the sample method?

Reply:
My bad, I had the other case in my mind (constraint columns being dropped). You are totally right.